src/libraries/Microsoft.Extensions.Logging/tests/Common/Console/TestConsole.cs

```diff
@@ -25,6 +25,92 @@ public TestConsole(ConsoleSink sink)
 
         public ConsoleColor? ForegroundColor { get; private set; }
 
+        public void Write(string message)
```

Comment:
NOTE:

This TestConsole class helps test ConsoleLogger functionality, which is why I just copy-pasted of the parser code as I was not interested to test that part here. (Used by ConsoleLoggerTests class containing existing tests).

- [ ] I could probably do better here to mock or something. I'll see if I can improve this